internal/backendstore/backendstore.go

```diff
@@ -21,21 +21,64 @@ func NewBackendStore() *BackendStore {
 	}
 }
 
+func (b *BackendStore) GetBackendClient(backendName string) *s3.Client {
+	b.mu.RLock()
```

Comment:
Does it make sense to merge these functions with `backend` struct to save one lock? In this way we have two locks, just to be programatically correct. We won't use `backend` without these helpers. WDYT?

Reply:
Could you elaborate on this a bit more please? I'm not sure I understand what this would look like...

Reply:
Currently we call `GetBackendClient` then lock `b.mu.RLock()` then call `getBackendClient` then we lock again `be.mu.RLock()`. But we never use `getBackendClient` only from `GetBackendClient` so maybe if we should combine the functionality into one, we should have only one lock. WDYT?

Reply:
Ah yes, sorry. Makes sense :+1: